src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImportOptions.java

```diff
@@ -39,4 +45,35 @@ public String getImportLocation()
    {
       return importLocation_;
    }
+   
+   public String getCode()
+   {
+      return code_;
+   }
+   
+   public int getMaxRows()
+   {
+      return maxRows_;
+   }
+   
+   public void setCode(String code)
+   {
+      code_ = code;
+   }
+   
+   public void setMaxRows(int maxRows)
+   {
+      maxRows_ = maxRows;
+   }
+   
+   public JSONObject toJSONObject()
```

Comment:
if an object will always be serialized to JSON then I'll often just implement entirely as a native JS object via native methods. I'll also typically give it a static create method that is basically a JNSI constructor. If you always create the object with all the fields then you could remove the set* methods once you have the create method.